src/kernel-install/ukify.py

```diff
@@ -21,6 +21,16 @@
 def guess_efi_arch():
     arch = os.uname().machine
     efi_arch = efi_arches[arch]
+
+    if arch == 'x86_64':
+        try:
+            size = open('/sys/firmware/efi/fw_platform_size').read().strip()
+        except FileNotFoundError:
+            pass
+        else:
+            if int(size) == 32:
+                efi_arch = 'ia32'
+
```

Comment:
wouldn't it make more sense to have some dictionary for this mapping? that maps native arch, and reported fw_platform_size value and maps it to another arch? i.e. even if the map will only have a single entry it sounds nicer to separate the code that reads the platform size from the table that maps it.